### ISSUE TITLE
End-to-end simple container workflow

### ISSUE THREAD

**Issue by danieldreier** [hashicorp/terraform-cdk#732]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
This is a draft! Let's talk before starting work on this.

The purpose of this project is to design a good end-to-end workflow for building and deploying containerized applications to popular cloud providers. This is simultaneously a documentation and design exercise. In the first iteration, the documentation should describe how to implement these workflows using CDK for Terraform as it exists today.

While writing that documentation, consider what else we could do to simplify the workflow and make it more idiomatic in each language and cloud provider. File enhancement requests, bugs, and link them back to this issue.

Use case: user wants to deploy a simple containerized application without having to stand up or operate other infrastructure

#### Desired workflow
* build a container
* push it to the cloud provider's native container registry service
* run it in AWS ECS, Azure Container Instances, and (if time permits) Google Cloud Run
* Set up load balancers, security groups, etc as needed to allow inbound HTTP access
* CDK for Terraform code examples should be implemented in TypeScript, Python, and Go

#### Stretch goals
As time permits, more advanced permutations would also be helpful, such as:
* deploying and configuring persistence (db, storage buckets) including setting up authentication to the persistence
* separately deploying a toy backend API and front-end as separate containers
* Provide a clear developer workflow for publishing to development/staging/production environments
* When the deploy completes, show (using an output?) the URL the web application is available at

### Deliverables
The immediate practical output will be end-to-end documentation published in this GitHub repository. File any ideas you have for improving it as enhancement requests linked to this issue for ease of tracking.

**Comment by RichiCoder1** [comment]
> AWS ECS

AWS App Runner would probably be the closest equivalent to Cloud Run & ACI and would probably a simpler example.

Excited to see this come to fruition though! A lot of the best parts of the CDK is the high level (L2?) constructs.

**Comment by DanielMSchmidt** [comment]
@RichiCoder1 Thanks for pointing towards App Runner, that service was new to me. It seems really simple to a point where it would make cdk patterns hard to see, therefore I'll try the ECS route first :)

**Comment by danieldreier** [comment]
This has been addressed with https://github.com/hashicorp/docker-on-aws-ecs-with-terraform-cdk-using-typescript and https://github.com/hashicorp/kubernetes-on-gcp-with-terraform-cdk. Thanks @DanielMSchmidt!
